crates/bevy_wgpu/src/renderer/wgpu_render_resource_context.rs

```diff
@@ -480,7 +482,23 @@ impl RenderResourceContext for WgpuRenderResourceContext {
                         }
                         RenderResourceBinding::Buffer { buffer, range, .. } => {
                             let wgpu_buffer = buffers.get(&buffer).unwrap();
-                            wgpu::BindingResource::Buffer(wgpu_buffer.slice(range.clone()))
+
+                            let offset = match range.start_bound() {
+                                Bound::Included(&bound) => bound,
+                                Bound::Excluded(&bound) => bound + 1,
+                                Bound::Unbounded => 0,
+                            };
+                            let size = match range.end_bound() {
+                                Bound::Included(&bound) => BufferSize::new(bound + 1 - offset),
+                                Bound::Excluded(&bound) => BufferSize::new(bound - offset),
+                                Bound::Unbounded => None,
+                            };
+
+                            wgpu::BindingResource::Buffer {
+                                buffer: wgpu_buffer,
+                                offset,
+                                size,
+                            }
```

Comment:
Not sure if this is a good way to do this. With gfx-rs/wgpu-rs#523 we've got a new method .as_entire_binding(), but it takes the entire buffer as a biding resource as opposed to a range of the buffer.

This code was copied from `wgpu::range_to_offset_size` but it is private.

Reply:
the current code looks good to me!